Mathlib/RingTheory/Localization/Algebra.lean

```diff
@@ -0,0 +1,161 @@
+/-
+Copyright (c) 2024 Christian Merten. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Christian Merten
+-/
+import Mathlib.Algebra.Module.LocalizedModule
+import Mathlib.RingTheory.Ideal.Maps
+import Mathlib.RingTheory.Localization.Basic
+import Mathlib.RingTheory.Localization.Ideal
+
+/-!
+# Localization of algebra maps
+
+In this file we provide constructors to localize algebra maps. Also we show that
+localization commutes with taking kernels for ring homomorphisms.
+
+## Implementation detail
+
+The proof that localization commutes with taking kernels does not use the result for linear maps,
+as the translation is currently tedious and can be unified easily after the localization refactor.
+
+-/
+
+universe u u' v v' w w'
+
+variable {R S P : Type*} (Q : Type*) [CommSemiring R] [CommSemiring S] [CommSemiring P]
+  [CommSemiring Q]
+  {M : Submonoid R} {T : Submonoid P}
+  [Algebra R S] [Algebra P Q] [IsLocalization M S] [IsLocalization T Q]
+  (g : R →+* P)
+
+variable (M S) in
+/-- The span of `I` in a localization of `R` at `M` is the localization of `I` at `M`. -/
+instance Algebra.idealMap_isLocalizedModule (I : Ideal R) :
+    IsLocalizedModule M (Algebra.idealMap I (S := S)) where
+  map_units x := by
+    rw [isUnit_iff_exists]
+    have hu : IsUnit (algebraMap R S x) := IsLocalization.map_units S x
+    let φ : Module.End R
+        (I.map (algebraMap R S)) := {
+      toFun := fun y ↦ ⟨hu.unit⁻¹ * y, by simp⟩
+      map_add' := fun x y ↦ by simp [mul_add]
+      map_smul' := fun a x ↦ by simp
+    }
+    refine ⟨φ, ?_, ?_⟩
+    · ext y
+      simp [φ, Algebra.smul_def, ← mul_assoc]
+    · ext y
+      simp [φ, Algebra.smul_def, ← mul_assoc]
+  surj' y := by
+    obtain ⟨x, hx⟩ := (IsLocalization.mem_map_algebraMap_iff M S).mp y.property
+    use x
+    apply Subtype.ext
+    simp [Submonoid.smul_def, Algebra.smul_def, mul_comm, hx]
+  exists_of_eq h := by
+    obtain ⟨c, hc⟩ := IsLocalization.exists_of_eq (M := M) (congrArg Subtype.val h)
+    exact ⟨c, Subtype.ext hc⟩
+
+lemma IsLocalization.ker_map (hT : Submonoid.map g M = T) :
```

Comment:
This should go near `IsLocalization.map`

Reply:
This needs `Mathlib.RingTheory.Localization.Ideal` (and the `Localization.Basic` file does not even know kernels of ring homs). The `Ideal` file does not currently talk about `IsLocalization.map`, should I still move it there?

Reply:
I think putting it in the `Ideal` file makes sense.

Reply:
Okay, I leave it here then.